pkg/virt-controller/services/template.go

```diff
@@ -1547,44 +1546,64 @@ func getIfaceByName(vmi *v1.VirtualMachineInstance, name string) *v1.Interface {
 	return nil
 }
 
-func getCniAnnotations(vmi *v1.VirtualMachineInstance) (cniAnnotations map[string]string, err error) {
-	ifaceListMap := make([]map[string]string, 0)
-	cniAnnotations = make(map[string]string, 0)
+func getMultusCNIAnnotation(vmi *v1.VirtualMachineInstance) (string, error) {
+	var ifaceListMap []map[string]string
+
+	relevantNetworks := filterMultusNonDefaultNetworks(vmi)
+	for i, network := range relevantNetworks {
+		iface := getIfaceByName(vmi, network.Name)
+		namespace, networkName := getNamespaceAndNetworkName(vmi, network.Multus.NetworkName)
+		serializedInterface := mapifyMultusInterface(iface, networkName, namespace, i)
+		ifaceListMap = append(ifaceListMap, serializedInterface)
+	}
+
+	if len(ifaceListMap) > 0 {
+		ifaceJsonString, err := serializeMultusInterfaces(ifaceListMap)
+		if err != nil {
+			return "", err
+		}
+		return ifaceJsonString, nil
+	}
+	return "", nil
+}
+
+func serializeMultusInterfaces(ifaceListMap []map[string]string) (string, error) {
+	ifaceJsonString, err := json.Marshal(ifaceListMap)
+	if err != nil {
+		return "", fmt.Errorf("Failed to create JSON list from CNI interface map %s", ifaceListMap)
+	}
+	return string(ifaceJsonString), nil
+}
 
-	next_idx := 0
+func mapifyMultusInterface(multusInterface *v1.Interface, networkName string, namespace string, interfaceSuffix int) map[string]string {
+	serializedMultusIface := map[string]string{
+		"name":      networkName,
+		"namespace": namespace,
+		"interface": fmt.Sprintf("net%d", interfaceSuffix+1),
+	}
+
+	if multusInterface != nil && multusInterface.MacAddress != "" {
```

Comment:
In what case the interface can be nil? How can this be valid?

Reply:
We never check what we get out of the `getIfaceByName`. 

Just keeping the old behavior.